code/modules/hydroponics/unique_plant_genes.dm

```diff
@@ -595,28 +614,55 @@
 /datum/plant_gene/trait/gas_production/proc/stop_gas(datum/source)
 	SIGNAL_HANDLER
 
-	stinky_seed = null
-	home_tray = null
 	STOP_PROCESSING(SSobj, src)
 
 /*
  * If the conditions are acceptable and the potency is high enough, release miasma into the air.
  */
 /datum/plant_gene/trait/gas_production/process(delta_time)
-	if(!stinky_seed || !home_tray || stinky_seed.loc != home_tray)
+	if(!home_tray || !stinky_seed)
```

Comment:
these are weakrefs, you're going to want to resolve them and then check for the existence of the resolved references